src/util/check.h

```diff
@@ -93,6 +93,7 @@ constexpr T&& inline_assertion_check(LIFETIMEBOUND T&& val, [[maybe_unused]] con
  * - For fatal errors, use Assert().
  * - For non-fatal errors in interactive sessions (e.g. RPC or command line
  *   interfaces), CHECK_NONFATAL() might be more appropriate.
+ *   Note that the assumption is always fully evaluated - even in non-debug builds.
```

Comment:
Since `Assume` is a macro, it could have been written differently to avoid parameter evaluation (similarly to `LogDebug`), so I think it needs a comment.
But I'm fine doing it in a different PR is this is remotely controversial.

Reply:
Migrated it to https://github.com/bitcoin/bitcoin/pull/31528

Reply:
Thank you for clarifying.
But given that we've bumped into this multiple times, do you think https://github.com/bitcoin/bitcoin/pull/31528 isn't a valid change?